File: app/src/main/java/com/theandroiddev/mywins/successslider/SuccessImageLoader.java
package com.theandroiddev.mywins.successslider;

import com.theandroiddev.mywins.data.models.Success;
import com.theandroiddev.mywins.data.models.SuccessImage;
import com.theandroiddev.mywins.data.repositories.SuccessesRepository;

import java.util.ArrayList;

/**
 * Created by jakub on 12.11.17.
 */

public class SuccessImageLoader implements SuccessSliderContract.SuccessImageLoader {
    private SuccessesRepository successesRepository;

    @Override
    public ArrayList<SuccessImage> getSuccessImages(String id) {
        return successesRepository.getSuccessImages(id);
    }

    @Override
    public void setRepository(SuccessesRepository successesRepository) {

        this.successesRepository = successesRepository;
    }

    @Override
    public Success getSuccess(String id) {
        return successesRepository.getSuccess(id);
    }
}

File: app/src/main/java/com/theandroiddev/mywins/success_slider/SuccessImageLoader.kt
package com.theandroiddev.mywins.success_slider

import com.theandroiddev.mywins.data.models.Success
import com.theandroiddev.mywins.data.models.SuccessImage
import java.util.*

interface SuccessImageLoader {

    abstract fun getSuccessImages(id: Long): ArrayList<SuccessImage>

    abstract fun getSuccess(id: Long): Success

    abstract fun openDB()

    abstract fun closeDB()
}